src/components/Trade/TradeInput.tsx

```diff
@@ -218,35 +241,31 @@ export const TradeInput = () => {
     })
   }, [
     buyTradeAsset?.asset?.assetId,
+    swapperDispatch,
     quote,
     sellAssetBalanceCrypto,
     sellFeeAsset,
-    sellTradeAsset?.asset,
+    sellTradeAsset,
     setTradeAmountsRefetchData,
-    setValue,
   ])
 
-  const onSubmit = useCallback(
-    async (values: TradeState<KnownChainIds>) => {
-      setIsLoading(true)
-      moduleLogger.info(values, 'debugging logger')
-      try {
-        const isApproveNeeded = await checkApprovalNeeded()
-        if (isApproveNeeded) {
-          history.push({ pathname: TradeRoutePaths.Approval })
-          return
-        }
-        const trade = await getTrade()
-        setValue('trade', trade)
-        history.push({ pathname: TradeRoutePaths.Confirm })
-      } catch (e) {
-        moduleLogger.error(e, 'onSubmit error')
-      } finally {
-        setIsLoading(false)
+  const onSubmit = useCallback(async () => {
+    setIsLoading(true)
+    try {
+      const isApproveNeeded = await checkApprovalNeeded()
```

Comment:
Not introduced in this PR but if you want to LSP rename this you get a 🎖️ 
```suggestion
      const isApprovalNeeded = await checkApprovalNeeded()
```

Reply:
Nice! https://github.com/shapeshift/web/pull/3927/commits/d2886a3f5ab889ab5afa4e3886c1d4d0872841c8